tests/e2e/bindings/bindings_test.go

```diff
@@ -96,21 +99,32 @@ func TestBindings(t *testing.T) {
 	for _, mes := range testMessages {
 		req.Messages = append(req.Messages, messageData{Data: mes, Operation: "create"})
 	}
-
 	body, err := json.Marshal(req)
 	require.NoError(t, err)
 
-	_, err = utils.HTTPPost(fmt.Sprintf("%s/tests/send", outputExternalURL), body)
-	require.NoError(t, err)
+	// act
+	postAssertHTTPStatus(t, fmt.Sprintf("%s/tests/send", outputExternalURL), body, http.StatusOK)
 
-	// This delay allows all the messages to reach corresponding input bindings
+	// This delay allows all the messages to reach corresponding input bindings.
 	time.Sleep(bindingPropagationDelay * time.Second)
 
-	resp, err := utils.HTTPPost(fmt.Sprintf("%s/tests/get_received_topics", inputExternalURL), nil)
-	require.NoError(t, err)
+	// assert
+	resp := postAssertHTTPStatus(t, fmt.Sprintf("%s/tests/get_received_topics", inputExternalURL), nil, http.StatusOK)
 
 	var decodedResponse receivedTopicsResponse
 	err = json.Unmarshal(resp, &decodedResponse)
 	require.NoError(t, err)
-	require.Equal(t, testMessages, decodedResponse.ReceivedMessages)
+
+	// Only the first message fails, all other messages are successfully consumed.
+	// nine messages succeed.
+	require.Equal(t, testMessages[1:], decodedResponse.ReceivedMessages)
+	// one message fails.
+	require.Equal(t, testMessages[0], decodedResponse.FailedMessage)
+}
+
+func postAssertHTTPStatus(t *testing.T, url string, data []byte, status int) []byte {
```

Comment:
`httpPostWithAssert` or similar sounds better than `postAssertHTTPStatus`. what do you think?

Reply:
it is specifically asserting no error and only HTTP status .... so only went with postAssert`HTTPStatus`